includes/server/llms-rest-server-functions.php

```diff
@@ -42,52 +43,250 @@ function llms_rest_authorization_required_error( $message = '', $check_authentic
  *
  * @since 1.0.0-beta.1
+ * @since [version] Use WP_Http constant for the error status.
  *
  * @param string $message Optional. The custom error message. Default empty string.
  *                        When no custom message is provided a predefined message will be used.
  * @return WP_Error
  */
 function llms_rest_not_found_error( $message = '' ) {
 	$message = ! $message ? __( 'The requested resource could not be found.', 'lifterlms' ) : $message;
-	return new WP_Error( 'llms_rest_not_found', $message, array( 'status' => 404 ) );
+	return new WP_Error( 'llms_rest_not_found', $message, array( 'status' => WP_Http::NOT_FOUND ) ); // 404
 }
 
 /**
  * Return a WP_Error for a 500 Internal Server Error.
  *
  * @since 1.0.0-beta.1
+ * @since [version] Use WP_Http constant for the error status.
  *
  * @param string $message Optional. Custom error message. When none provided a predefined message is used.
  * @return WP_Error
  */
 function llms_rest_server_error( $message = '' ) {
 	$message = ! $message ? __( 'Internal Server Error.', 'lifterlms' ) : $message;
-	return new WP_Error( 'llms_rest_server_error', $message, array( 'status' => 500 ) );
+	return new WP_Error( 'llms_rest_server_error', $message, array( 'status' => WP_Http::INTERNAL_SERVER_ERROR ) ); // 500.
 }
 
 /**
- * Validate submitted array of integers is an array of real real user ids.
+ * Checks whether or not the passed object is an error of missing required authorization
+ *
+ * @since [version]
+ *
+ * @param WP_Error $wp_error The WP_Error object to check.
+ * @return boolean
+ */
+function llms_rest_is_authorization_required_error( $wp_error ) {
+	return ! empty( array_intersect( llms_rest_get_all_error_statuses( $wp_error ), array( WP_Http::FORBIDDEN, WP_Http::UNAUTHORIZED ) ) );
+}
+
+/**
+ * Checks whether or not the passed object is a bad request error
+ *
+ * @since [version]
+ *
+ * @param WP_Error $wp_error The WP_Error object to check.
+ * @return boolean
+ */
+function llms_rest_is_bad_request_error( $wp_error ) {
+	return in_array( WP_Http::BAD_REQUEST, llms_rest_get_all_error_statuses( $wp_error ), true );
+}
+
+/**
+ * Checks whether or not the passed object is a not found error
+ *
+ * @since [version]
+ *
+ * @param WP_Error $wp_error The WP_Error object to check.
+ * @return boolean
+ */
+function llms_rest_is_not_found_error( $wp_error ) {
+	return in_array( WP_Http::NOT_FOUND, llms_rest_get_all_error_statuses( $wp_error ), true );
+}
+
+/**
+ * Checks whether or not the passed object is an internal server error
+ *
+ * @since [version]
+ *
+ * @param WP_Error $wp_error The WP_Error object to check.
+ * @return boolean
+ */
+function llms_rest_is_server_error( $wp_error ) {
+	return in_array( WP_Http::INTERNAL_SERVER_ERROR, llms_rest_get_all_error_statuses( $wp_error ), true );
+}
+
+/**
+ * Returns all the error statuses of a WP_Error
+ *
+ * @since [version]
+ *
+ * @param WP_Error $wp_error The WP_Error object.
+ * @return array
```

Comment:
This will be an array of error code integers, right? EG: `array( 404, 403 )` or something, right?

```suggestion
 * @return int[]
```